awex-core/src/main/java/com/raycoarana/awex/AwexPromise.java

```diff
@@ -686,6 +689,14 @@ public void onCancel() {
         return promise;
     }
 
+    @Override
+    public <Result_Out, Progress_Out> Promise<Result_Out, Progress_Out> pipe(
```

Comment:
I don't like the idea of adding a new pipe() method, I think you could implement this using the current pipe() method, providing a Promise to it. You just then need to add a fail callback to that promise.